### ISSUE TITLE
TypeError in key-spacing rule when using object rest spread

### ISSUE THREAD

**Issue by zaygraveyard** [eslint/eslint#5724]
**What version of ESLint are you using?**
`v2.5.3`

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```json
{
  "parserOptions": {
    "ecmaVersion": 6,
    "ecmaFeatures": {
      "experimentalObjectRestSpread": true
    }
  },
  "rules": {
    "key-spacing": [1, {"align": "colon"}]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**
```js
var x = {...object};
```

**What did you expect to happen?**
The linting passes with no errors.

**What actually happened? Please include the actual, raw output from ESLint.**
```
$ eslint test.js                                                                                                                                                                                                                         
Cannot read property 'range' of undefined
TypeError: Cannot read property 'range' of undefined
    at getPropertyWhitespace (/home/zkhoury/tmp/eslint-tests/node_modules/eslint/lib/rules/key-spacing.js:242:25)
    at verifySpacing (/home/zkhoury/tmp/eslint-tests/node_modules/eslint/lib/rules/key-spacing.js:333:22)
    at verifyListSpacing (/home/zkhoury/tmp/eslint-tests/node_modules/eslint/lib/rules/key-spacing.js:349:13)
    at EventEmitter.ObjectExpression (/home/zkhoury/tmp/eslint-tests/node_modules/eslint/lib/rules/key-spacing.js:362:21)
    at emitOne (events.js:77:13)
    at EventEmitter.emit (events.js:169:7)
    at NodeEventGenerator.enterNode (/home/zkhoury/tmp/eslint-tests/node_modules/eslint/lib/util/node-event-generator.js:42:22)
    at CodePathAnalyzer.enterNode (/home/zkhoury/tmp/eslint-tests/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:608:23)
    at CommentEventGenerator.enterNode (/home/zkhoury/tmp/eslint-tests/node_modules/eslint/lib/util/comment-event-generator.js:98:23)
    at Controller.traverser.traverse.enter (/home/zkhoury/tmp/eslint-tests/node_modules/eslint/lib/eslint.js:869:36)
```

**Comment by gyandeeps** [comment]
Based on your config you are using default (espree) parser. Since object spread is not part of ES6 or ES7 that why we dont support it out of the box. 
Link: https://github.com/eslint/eslint#what-about-ecmascript-6-support

For your purpose please use `babel-eslint`.

**Comment by zaygraveyard** [comment]
I remembered that I had previously found a similar bug #4763.
And after further investigation I found that I had missed a branch of the `if` which is causing this error.
Would gladly do a PR with the fix if needed.

**Comment by btmills** [comment]
Verified on master. Yes @zaygraveyard, please do send a PR!

**Comment by zaygraveyard** [comment]
PR at #5729
